nimble/host/include/host/ble_audio_broadcast.h

```diff
@@ -0,0 +1,271 @@
+#define H_BLE_AUDIO_BROADCAST_
+
+#include <stdint.h>
+#include "host/ble_gap.h"
+#include "host/ble_iso.h"
+#include "host/ble_audio_common.h"
+
+struct ble_broadcast_create_params {
+    /** BASE to broadcast */
+    struct ble_audio_base *base;
+
+    /** Parameters used to configure Extended advertising */
+    struct ble_gap_ext_adv_params *extended_params;
+
+    /** Parameters used to configure Periodic advertising */
+    struct ble_gap_periodic_adv_params *periodic_params;
+
+    /** Broadcast name - null terminated.
+     * Set NULL to not include in advertising
+     */
+    const char *name;
+
+    /** Advertising instance */
+    uint8_t adv_instance;
+
+    /** BIG parameters */
+    struct ble_iso_big_params *big_params;
+
+    /** Additional data to include in Extended Advertising  */
+    uint8_t *svc_data;
+
+    /** Additional data length  */
+    uint16_t svc_data_len;
+};
+
+struct ble_broadcast_update_params {
+    /** Broadcast name - null terminated.
+     * Set NULL to not include in advertising
+     */
+    const char *name;
+
+    /** Advertising instance */
+    uint8_t adv_instance;
+
+    /** Additional data to include in Extended Advertising  */
+    uint8_t *svc_data;
+
+    /** Additional data length  */
+    uint16_t svc_data_len;
+
+    /** Broadcast ID */
+    uint32_t broadcast_id;
+};
+
+typedef int ble_audio_broadcast_destroy_fn(struct ble_audio_base *base,
+                                           void *args);
+
+/** BASE configuration describing broadcast advertisement */
+struct ble_broadcast_base_config {
+    /** Advertising instance used by broadcast */
+    uint8_t adv_instance;
+
+    /** Pointer to BASE configuration */
+    struct ble_audio_base *base;
+
+    /** BIG parameters */
+    struct ble_iso_big_params *big_params;
+
+    /**
+     * Optional callback associated with broadcasting instance, called on
+     * destroying broadcast
+     */
+    ble_audio_broadcast_destroy_fn *destroy_cb;
+
+    /** Pointer to args for `destroy_cb`  */
+    void *args;
+};
+
+/**
+ * @brief Create Broadcast Audio Source Endpoint and configure advertising
+ * instance
+ *
+ * This function configures advertising instance for extended and periodic
+ * advertisements to be ready for broadcast with BASE configuration.
+ *
+ * @param[in] params            Pointer to a `ble_broadcast_base_params`
+ *                              structure that defines BASE, extended
+ *                              advertising and periodic advertising
+ *                              configuration.
+ * @param[out] config_out       Pointer to a `ble_broadcast_base_config`
+ *                              structure to return configuration of created
+ *                              BASE advertisement.
+ * @param[in] destroy_cb        Optional callback to be called when BASE
+ *                              advertisement is destroyed.
+ * @param[in] args              Optional arguments to be passed to `destroy_cb`
```

Comment:
I'm planning to add sample that acts as usb soundcard for pc, where all of this API will be used. But that requires some additional code, outside of scope of this PR, like LC3. For now I added shell commands to test this code. I think it would be nice to add unittests too, when this API is semi-established I'll add them I think

Reply:
I mean args for destroy_cb :)

Reply:
Anything you need, it's a void pointer, app will cast it back to type it needs. Intention is, that for applications using dynamic memory allocation (like mempool) for BASE, BIG subgroups, BISes etc. this could be used as a pointer to to these things, and then app can free these resources. You can see that in `btshell_broadcast_destroy_fn`